### ISSUE TITLE
ASM specific cmdlets no longer work

### ISSUE THREAD

**Issue by pjirsa** [Azure/azure-powershell#1777]
Is this by design? After upgrading from 0.9.x to 1.2.0 I cannot use any of the ASM cmdlets. I can run add-azureaccount and authenticate. But, no subscriptions come back as a result which causes all of the ASM commands to not work.

**Comment by markcowl** [comment]
@pjirsa Add-AzureAccount is now only for ASM usage, so it will only return subscriptions for which you are an admin or co-admin.  For using ARM cmdlets, you must run Add-AzureRmAccount, which uses RBAC to decide which subscriptions you have access to.

If you think there are subscriptions for which you are an admin or co-admin that are not being returned by Add-AzureAccount, please run the cmdlet in Debug mode and we can debug through the issue.

**Comment by pjirsa** [comment]
Mark, thanks for the suggestion. I ran the following
```
add-azureaccount -tenant "<mytenant>.onmicrosoft.com" -Debug
```
I saw the get request and the returned list of subscriptions. However, calling get-azuresubscription returned nothing and none of the ASM commands work without a subscription context.

I was able to make it working by removing the -tenant switch and then manually selecting the subscription.

Maybe someone should look at why it doesn't work when using the -tenant switch.
